x/clp/keeper/msg_server.go

```diff
@@ -652,11 +674,30 @@ func (k msgServer) RemoveLiquidityUnits(goCtx context.Context, msg *types.MsgRem
 		pool.NativeAssetBalance.String(), pool.ExternalAssetBalance.String(), lp.LiquidityProviderUnits.String(),
 		msg.WithdrawUnits)
 
+	normalizationFactor, adjustExternalToken := k.GetNormalizationFactor(eAsset.Decimals)
+	extRowanValue, err := CalculateWithdrawalRowanValue(withdrawExternalAssetAmount, types.GetSettlementAsset(), pool, normalizationFactor, adjustExternalToken, pmtpCurrentRunningRate)
+	if err != nil {
+		return nil, err
+	}
+
 	err = k.Keeper.UseUnlockedLiquidity(ctx, lp, lp.LiquidityProviderUnits.Sub(lpUnitsLeft), false)
 	if err != nil {
 		return nil, err
 	}
 
+	futurePool := pool
+	futurePool.NativeAssetBalance = futurePool.NativeAssetBalance.Sub(withdrawNativeAssetAmount)
+	futurePool.ExternalAssetBalance = futurePool.ExternalAssetBalance.Sub(withdrawExternalAssetAmount)
+	if k.GetMarginKeeper().CalculatePoolHealth(&futurePool).GT(k.GetMarginKeeper().GetRemovalQueueThreshold(ctx)) {
+		k.QueueRemoval(ctx, &types.MsgRemoveLiquidity{
```

Comment:
@timlind I see this method is used in both `RemoveLiquidity` and `RemoveLiquidityByUnits` although the `msg` type used and that is later passed to emit the event use the same for both, would that be a problem to have accurate event data?